core/crypto/typed_key_storage/typed_key_storage_impl.hpp

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_TYPED_KEY_STORAGE_IMPL_HPP
+#define KAGOME_TYPED_KEY_STORAGE_IMPL_HPP
+
+#include "crypto/ed25519_types.hpp"
+#include "crypto/sr25519_types.hpp"
+#include "crypto/typed_key_storage.hpp"
+
+namespace kagome::crypto::storage {
+  class TypedKeyStorageImpl : public TypedKeyStorage {
+   public:
+    EDKeys getEdKeys(KeyTypeId key_type) override;
+
+    SRKeys getSrKeys(KeyTypeId key_type) override;
+
+    void addEdKeyPair(KeyTypeId key_type,
+                      const ED25519Keypair &key_pair) override;
+
+    void addSrKeyPair(KeyTypeId key_type,
+                      const SR25519Keypair &key_pair) override;
+
+    boost::optional<ED25519Keypair> findEdKey(
```

Comment:
Optional:
Think about to raname method by full algo-name, i.e. `findEd25519Key`.